IHP/ModelSupport.hs

```diff
@@ -921,6 +973,16 @@ instance ToField value => ToField [value] where
 instance (FromField value, Typeable value) => FromField [value] where
     fromField field value = PG.fromPGArray <$> (fromField field value)
 
+-- | Wraps a list of actions to be used as a Tuple, useful e.g for matching composite keys
+-- >>> toField (ActionTuple [ PG.Escape "myId" ])
+-- Many [Plain "(",Escape "myId",Plain ")"]
+--
+--   Analogous to PGArray from postgres-simple
+newtype ActionTuple = ActionTuple [Action]
+
+instance ToField ActionTuple where
+  toField (ActionTuple actions) = PG.Many ([PG.Plain "("] <> intersperse (PG.Plain ",") actions <> [PG.Plain ")"])
+
```

Comment:
Could we simplify this further by getting rid of the newtype and then inlining the definition into `primaryKeyConditionForId` or some other function? This feels like one level of indireciton we might not need (but could be wrong here)

Reply:
That would mean that `primaryKeyConditionForId` no longer returns seperate fields for each column...

Which is fine, I guess? Makes it a little bit less flexible, but so far that flexibility isn't needed...

Reply:
I'll quickly take a look if this has any other implications

Reply:
I think this is a good idea

The primaryKeyCondition would be directly generated by the SchemaCompiler, and the `Table` instances will look like:
```haskell
instance Table CompositeTagging where
    columnNames = ["post_id", "tag_id"]
    primaryKeyColumnNames = ["post_id", "tag_id"]
    primaryKeyConditionForId (Id (postId, tagId)) = PG.Many ([PG.Plain "(", toField postId, PG.Plain "," toField tagId, PG.Plain ")"])

instance Table User where
    columnNames = ["id", "name"]
    primaryKeyColumnNames = ["id"]
    primaryKeyConditionForId (Id id) = PG.Many ([PG.Plain "(", toField id, PG.Plain ")"])

```

I think we could even go so far to make a special case for single-column pks, so that they look like

```haskell
instance Table User where
    columnNames = ["id", "name"]
    primaryKeyColumnNames = ["id"]
    primaryKeyConditionForId (Id id) = toField id
```

which would be rather desirable, but I'll have to look a little bit closer to find out if that would break things